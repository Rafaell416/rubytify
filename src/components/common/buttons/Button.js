import React from 'react'
import PropTypes from 'prop-types'
import { Heading2 } from '../../typography'
import { colors } from '../../../styles'
import { Link } from 'react-router-dom'

function Button ({ 
  children, 
  link, 
  outline, 
  to,
  }) {
  const styles = {
    buttonContainer: {
      backgroundColor: outline ? 'transparent' : colors.primary,
      height: 40,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: 150,
      border: outline ? '.5px solid' : 'none',
      borderColor: colors.primary,
      borderRadius: 25,
      cursor: 'pointer',
      outline: 'none',
      marginLeft: 5,
      marginRight: 5
    }
  }

    if (link) {
      return (
        <Link style={{ textDecoration: 'none' }} to={to}>
          <button style={styles.buttonContainer}>
            <Heading2>
              {children}
            </Heading2>
          </button>
        </Link>
      )
    } else {
      return (
        <button style={styles.buttonContainer}>
          <Heading2>
            {children}
          </Heading2>
        </button>
      )
    }
} 

Button.propTypes = {
  color: PropTypes.string,
  style: PropTypes.object,
  outline: PropTypes.bool
}

export default Button