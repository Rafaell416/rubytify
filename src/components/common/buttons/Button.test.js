import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

test('Testing <Button /> component', () => {
  const component = renderer.create(
    <Button>testing button</Button>
  )
  const instance = component.root

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(instance.children[0]._fiber.elementType).toBe('button')
})