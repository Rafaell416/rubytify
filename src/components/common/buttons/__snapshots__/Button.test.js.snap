// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Button /> component 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2EE59D",
      "border": "none",
      "borderColor": "#2EE59D",
      "borderRadius": 25,
      "cursor": "pointer",
      "display": "flex",
      "height": 40,
      "justifyContent": "center",
      "marginLeft": 5,
      "marginRight": 5,
      "outline": "none",
      "width": 150,
    }
  }
>
  <h2
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto, sans-serif",
        "margin": 0,
      }
    }
  >
    testing button
  </h2>
</button>
`;
